trunner/harnesses/micropython.py

```diff
@@ -0,0 +1,243 @@
+        text = text.replace(f"{EOL}########", "(.|\n)*")
+        text = text.replace("########", "(.|\n)*")
+
+        return text
+
+    @staticmethod
+    def expect(what, proc, timeout=5):
+        patterns = [what, pexpect.EOF, pexpect.TIMEOUT]
+        idx = proc.expect(patterns, timeout=timeout)
+
+        assert idx != 1, f"There was no \"{what}\" in the subprocess output"
+        assert idx != 2, f"Timeout occurred during waiting for \"{what}\""
+
+    # This function is needed for harness tests
+    @staticmethod
+    def get_before_output(proc):
+        """Function needed for harness tests."""
+
+        # In normal conditions proc.before returns a string
+        # While testing and communicating through socket proc.before returns a byte-object
+        out = proc.before
+
+        if not isinstance(out, str):
+            out = out.decode("utf-8")
+
+        return out
+
+
+class MicropythonStandardHarness:
+    """Class providing harness for getting test result from MicroPython test and comparing to the expected output"""
+
+    SPECIAL_TESTS = (
+        "micropython/meminfo.py",
+        "basics/bytes_compare3.py",
+        "basics/builtin_help.py",
+        "thread/thread_exc2.py",
+        "esp32/partition_ota.py",
+        "cmdline/cmd_parsetree.py",
+        "cmdline/cmd_showbc.py",
+        "cmdline/cmd_verbose.py",
+        "test/special_test.py"          # This one is for harness tests
+    )
+
+    def is_special_test(self, path_to_test):
+        """Some tests needs special handling - their expected output is a regex, not just string"""
+
+        p = pathlib.Path(path_to_test)
+        fragments = p.parts
+
+        if fragments[-2] + "/" + fragments[-1] in self.SPECIAL_TESTS:
+            return True
+        return False
+
+    def get_test_path_and_result(self, proc):
+        Tools.expect(PROMPT, proc, timeout=45)
+        program_output = Tools.get_before_output(proc)
+
+        index = program_output.find("\nError:")
+        assert index == -1, f"There was an error during program execution: {program_output}"
+
+        # Looking for a path to the test
+        index = re.search(r'/.+\.py', program_output)
+        assert index is not None, f"Can't find path to the test\nProgram output:\n{program_output}"
+        index = index.span()
+
+        test_path = program_output[index[0]+1:index[1]]
+
+        # After the path there is a program result
+        index = program_output.find('\n')
+        test_result = program_output[index+1:]
+
+        return (test_path, test_result)
+
+    def get_exp_output(self, test_path, proc):
+        expected_output_path = test_path + ".exp"
+
+        proc.sendline("cat %s" % expected_output_path)
+        Tools.expect(PROMPT, proc)
+
+        exp_file_cat = Tools.get_before_output(proc)
+
+        # exp_file_cat have a header "cat (path to the expected outpue)"
+        # removing the header
+        index = exp_file_cat.find('\n')
```

Comment:
Just send command end expect it afterwards. I mean `proc.expect(rf"cat {expected_output_path}{EOF}")` (it may not work in this form, look at some examples in `psh` directory)